src/pages/dataElements/fields/AggregationTypeField.tsx

```diff
@@ -51,19 +53,36 @@ export function AggregationTypeField() {
 
```

Comment:
Can't select the "options" lines because they were not changed. But I think it would be better to either get this using the `enum` from generated objects.
Right now, if a new `aggregationType` was added in the backend, you would be able to select it, but the validation would fail. 

I think ideally we should be able to create the "zod" schema using data from schema. But that makes it a bit more complex, because you would depend on state for the schema.

Reply:
> Right now, if a new aggregationType was added in the backend, you would be able to select it, but the validation would fail.

You mean in case the new type was added while the user is using the app already and then navigates to the form? I suppose the only way to work around that is to load the aggregation types from the backend rather than using the schema.

Right now, I don't see the advantage of using the zod schema anymore. But I guess we can discuss that separately as it's difficult to have a structured discussion in this comments section, let's continue this discussion on slack, I'll prepare a proper message with reasoning and thoughts

Reply:
I mean if a new AggregationType is added to the backend. You would generate and include this as an option here. But it would not pass the zod validation because that is static and not tied to the schema.

Wouldn't need to be added while user is in the app? I'm just saying this causes the validation and generated options to use different sources. I agree we can take another look at the zod approach.

Reply:
So, instead of removing zod entirely, I just changed `valueType` and `aggregationType` to `z.string()` in the schema. This will allow any new value that isn't reflected in the generated types